.github/workflows/build.yml

```diff
@@ -1,6 +1,6 @@
 name: Build dev environments
 on:
-  push:
+  pull_request:
```

Comment:
I have found myself in positions where I wanted to run some CI tests for a branch, without necessarily sending a PR. Or, when I have sent a PR already, and I've made a big variation that I don't want to force-push to a reviewed branch. To accommodate for this case, I think we can have something like this:

```
on:
  push:
    branches:
      - main
      - "test/**"
  pull_request:
```

That is, either run the CI tests on branches named `test/**` (useful only for devs) or `main` (happens whenever we merge a PR), or on pull requests that target any branch.  As long as we don't send PRs with feature branches named `tests/**`, we will not see duplicate runs. What do you think?

I think this way, there's no need for the `should-run` job as well.

Reply:
Thanks, I've updated the pull request accordingly!